src/Interceptor/Interceptor/EventInterceptor.php

```diff
@@ -84,7 +100,16 @@ public function supportsPrefix(Instance $instance): bool
 
     public function supportsSuffix(Instance $instance): bool
     {
-        return $this->supportsPrefix($instance);
+        if ($this->supportsPrefix($instance)) {
+            return true;
+        }
+
+        $useCaseInstance = array_filter(
+            $instance->getReflection()->getInterfaceNames(),
+            static fn (string $interfaceName) => str_ends_with($interfaceName, 'UseCase')
+        );
+
+        return !empty($useCaseInstance);
```

Comment:
I check that the class is a UseCase. i didn't wanted to create a link between ServiceProxy and UseCaseProxy in order to have the class name `OpenClassrooms/UseCase/Requestors/UseCase` so any class that implement an interface that end with `*/*/*UseCase` will do it,
if you find another way to target only UseCase I'm open to suggestion here 😅

Reply:
This will make all use cases auto implement the event attribute without needing to add the attribute, I'd say it's the target, but can be a bit dangerous as since UseCaseBundle exists in the project we might send the same event two times.

Reply:
Maybe we could have a dedicated interface for this purpose ? Something like `SendsGenericEvent` that would be implemented on the developer's side (OC).
Thus we would have our new (previously deprecated) UseCase interface that would extend both `Interceptable` and `SendsGenericEvent` ?

Btw, how relevant would it be to have a proper Interceptor for that ? My idea is to split responsibilities between our existing EventInterceptor (who only supports @Event annotation) and a GenericEventInterceptor, who would only supports proxies implementing SendsGenericEvent.

Reply:
Yep, we postponed the reflexion on this subject for now, and will discuss the possible options after merging everything 
- move the generic event into sdzv4 and creating a custom interceptor for it
- create a config for generic events using patterns ?